airflow/dags/unzip_and_validate_gtfs_schedule/unzip_gtfs_schedule.py

```diff
@@ -84,41 +84,42 @@ def summarize_zip_contents(zip: zipfile.ZipFile, at: str = ""):
         if zipfile.Path(zip, at=entry).is_dir():
             directories.append(entry)
     logging.info(f"Found files: {files} and directories: {directories}")
-    return files, directories
+    # the only valid case for any directory inside the zipfile is if there's exactly one and it's the only item at the root of the zipfile
+    # (in which case we can treat that directory's contents as the feed)
+    is_valid = (not directories) or (
+        (len(directories) == 1)
+        and ([item.is_dir() for item in zipfile.Path(zip, at="").iterdir()] == [True])
+    )
```

Comment:
I think you can simplify this; `not directories or (len(directories) == 1 and not files)`

Reply:
No, `files` has everything, not just at the root. `zipfile.Path(zip, at="").iterdir()]` only returns things that are at the root level in the zipfile. `files` will have *all* file type contents, regardless of where they fall in nested structure.